### PR TITLE
Now that MLIR supports floating point greater than double precision,

### PR DESCRIPTION
**Pull request by schweitzpgi**
switch lowering to use the new builtin types.

### CODE REVIEW

File: flang/test/Lower/real.f90 [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  REAL(4) :: x1
+  ! CHECK-DAG: %[[v1:.+]] = fir.load %arg0 : !fir.ref<f32>
+  ! CHECK-DAG: %[[v2:.+]] = fir.load %arg1 : !fir.ref<f32>
+  ! CHECK: %[[v3:.+]] = fir.addf %[[v1]], %[[v2]] : f32
+  real4 = x0 + x1
+  ! CHECK: return %{{.*}} : f32
+END FUNCTION real4
+
+! CHECK-LABEL: defreal
+REAL FUNCTION defreal(x0, x1)
```
Comment (vdonaldson) [inline flang/test/Lower/real.f90:39]: Isn't the default REAL choice subject to command line option values?

**Review by vdonaldson — APPROVED** [review]

File: flang/test/Lower/real.f90 (hunk 1 above)
Comment (schweitzpgi) [reply flang/test/Lower/real.f90:39]: Did we add such options to bbc?

Comment (vdonaldson) [reply flang/test/Lower/real.f90:39]: Whatever the current driver status is, default type options are common across compilers.  nvfortran has _-M[no]r8_, _-M[no]r8intrinsics[=float]_,  _-r4_, _-r8_.

Comment (schweitzpgi) [reply flang/test/Lower/real.f90:39]: I understand your point. This test is specifically written for `bbc` though.

**Pull request merged**